cns/service/main.go

```diff
@@ -918,11 +910,42 @@ func InitializeCRDState(ctx context.Context, httpRestService cns.HTTPService, cn
 		}
 	}()
 
+	clientset, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to build clientset")
+	}
+
+	var podInfoByIPProvider cns.PodInfoByIPProvider
+	if cnsconfig.InitializeFromCNI {
+		logger.Printf("Initializing from CNI")
+		podInfoByIPProvider, err = cnireconciler.NewCNIPodInfoProvider()
+		if err != nil {
+			return errors.Wrap(err, "failed to create CNI PodInfoProvider")
+		}
+	} else {
+		logger.Printf("Initializing from Kubernetes")
```

Comment:
Shouldnt we move this under retry loop to avoid crashloop backof not hitting and thus delaying cns startup in case apiserver is not ready

Reply:
I have moved the list pods inside the closure so that, if the apiserver is not ready, we also retry that during the podinfobyip call